builtin/logical/pki/path_config_crl.go

```diff
@@ -326,6 +212,10 @@ func (b *backend) pathCRLWrite(ctx context.Context, req *logical.Request, d *fra
 		config.UnifiedCRLOnExistingPaths = unifiedCrlOnExistingPathsRaw.(bool)
 	}
 
+	if maxCRLSizeRaw, ok := d.GetOk("max_crl_size"); ok {
+		config.MaxCRLSize = maxCRLSizeRaw.(int)
```

Comment:
Hmm, the problem is in storage, we'll unmarshall this as 0 if it's never been touched, and we want the default to take effect on upgrade.

Reply:
A few options

1. We leave it as is, and say setting the value to zero you get whatever default we now have
2. We explicitly forbid a value of 0 being sent to us, trapping it here, I like this the least
3. We use the default/missing value trick for JSON decoding by setting the field value passed into DecodeJSON with the default, that isn't overwritten if the json is missing the value, otherwise it is. 
   a. `result.MaxCRLSize = pki_backend.DefaultCrlConfig.MaxCRLSize` around line 645 in `getRevocationConfig`

Reply:
Saw #3, and what about 
4. Setting to 0 is basically ignored and doesn't change whatever the current value is.

Reply:
Or errors out on 0 or < -1

Reply:
> Or errors out on 0 or < -1

If we add this check only within the path_crl_config I'm good with that.